### DISCUSSION TITLE
[ 功能建议 ] 回滚工单简化审批流程

### DISCUSSION THREAD

**Discussion by LeoQuote** [hhyo/Archery#1585] [Ideas]
### 目前遇到的问题/使用障碍
回滚需要额外审批, 进程缓慢, 在数据量大(10W ,100W)的情况还会造成浏览器卡顿, 一般回滚需求十分紧急, 需要很快的回滚, 这种障碍可能会影响到业务正常运行.
### 希望如何解决/实现它
将工单回滚简化成工单的一个按钮, 只能由特定权限的人操作, 并提供回滚执行语句的sql文件下载.
### 其他信息
如果有其他类似的产品功能或者图片信息，可在此提交

**Comment by hhyo** [comment]
- 数据量大这个实际上会引出一个bug，直接内容传递不过去，可以考虑使用后端返回
- 关于回滚免批，可能有几种场景需要考虑，insert和delete的回滚直接操作就行，但是update语句，由于回滚语句在row full模式下是全字段备份，如果在此期间有新的操作导致非工单修改字段被更新，回滚就会造成数据错误，一般需要业务方进行确认。
ddl类似，可能预先需要应用服务回滚，否则应用可能报错

**Comment by LeoQuote** [comment]
一般回滚请求都是业务方提出的, 只需要如果涉及到10W等级的数据回滚, 确认也意义不大, 回滚只涉及到update过的数据, 如果需要回滚, 这里提供的方案已经是能做到的最好方案了.

回滚新起一个工单初衷是流程化(甩锅), 我觉得可以考虑将工单扩展一下, 新起三个状态,已提交回滚申请, 回滚中, 已回滚, 并且在日志中体现回滚的提交人, 执行人是谁. 

回滚申请流程可以简化一下, 只需申请人申请, 即可执行, 执行人沿用工单的执行人设置.

**Comment by hhyo** [comment]
对的，一般是业务方提出回滚，但是如果最终回滚过程出现脏数据，一般也要依靠DBA来协助修复（逃不掉），更新频率高的表直接依靠全量语句回滚，出脏数据还是很有可能的
简化回滚流程是可以的，减少中间过程，但是必须要有第二人确认，否则所有正常结束的工单，提交人都可以越过DBA直接回滚数据，虽然提交人不会这么做，但是确实存在风险

**Comment by yzypals** [comment]
回滚这种操作, 怎么能省掉审批呢?

**Comment by hhyo** [comment]
回滚语句下载已在 #518 实现，关于流程可以再探讨一下
